docs/DeveloperGuide.adoc

```diff
@@ -46,177 +39,186 @@ The following class plays an important role at the architecture level:
+[[Design-Responder]]
+=== Responder component
 
-[[Design-Logic]]
-=== Logic component
-
-[[fig-LogicClassDiagram]]
-.Structure of the Logic Component
-image::LogicClassDiagram.png[]
+[[fig-ResponderClassDiagram]]
+.Structure of the Responder Component
+image::ResponderClassDiagram.png[]
 
 *API* :
 link:{repoURL}/src/main/java/seedu/address/logic/Logic.java[`Logic.java`]
 
-.  `Logic` uses the `AddressBookParser` class to parse the user command.
-.  This results in a `Command` object which is executed by the `LogicManager`.
-.  The command execution can affect the `Model` (e.g. adding a person).
-.  The result of the command execution is encapsulated as a `CommandResult` object which is passed back to the `Ui`.
-.  In addition, the `CommandResult` object can also instruct the `Ui` to perform certain actions, such as displaying help to the user.
-
-Given below is the Sequence Diagram for interactions within the `Logic` component for the `execute("delete 1")` API call.
-
-.Interactions Inside the Logic Component for the `delete 1` Command
-image::DeleteSequenceDiagram.png[]
 
-NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
 
 [[Design-Model]]
 === Model component
 
 .Structure of the Model Component
-image::ModelClassDiagram.png[]
+image::ModelClassDiagram_FlashCardPro.png[]
 
-*API* : link:{repoURL}/src/main/java/seedu/address/model/Model.java[`Model.java`]
+*API* : link:{repoURL}/src/main/java/dream/fcard/model[`Model`]
 
-The `Model`,
+.Structure of the Model Component
+image::cards_class_diagram.png[]
 
-* stores a `UserPref` object that represents the user's preferences.
-* stores the Address Book data.
-* exposes an unmodifiable `ObservableList<Person>` that can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list change.
-* does not depend on any of the other three components.
 
-[NOTE]
-As a more OOP model, we can store a `Tag` list in `Address Book`, which `Person` can reference. This would allow `Address Book` to only require one `Tag` object per unique `Tag`, instead of each `Person` needing their own `Tag` object. An example of how such a model may look like is given below. +
- +
-image:BetterModelClassDiagram.png[]
+.`FlashCard`,
+
+* There are 3 types of cards used by our App, the `JavascriptCard`, `FrontBackCard` and `MultipleChoiceCard`.
+* `JavascriptCard` and `FrontBackCard` implements from `FlashCard`, an interface class.
+* `MultipleChoiceCard` inherits from `FrontBackCard`.
+* The interface class `FlashCard` also implements `JsonInterface` class to allow loading and saving to Json format.
+
+.FlashCard Priority
+Each FlashCard has an associated priority level, which the user can indicate his/her value of the importance of the card.
+
+The priority levels are described below:
```

Comment:
The priority level might be a bit ambiguous here. Is it represented as an enum of either `HIGH_PRIORITY` or `LOW_PRIORITY`? Or perhaps is it represented as an `Integer` in the range of `[1, 10]`.